README.md

```diff
@@ -1,24 +1,40 @@
+Note that IntelliJ has functionality to convert from JUnit 4 to 5. Consider trying that before this program. (This program was originally written in September 2016 when no IDEs had the functionality).
 
 # Pre-reqs
 Before starting out, the program assumes you have:
 * Updated your Maven POM/Groovy build file/whatever to reference the new JUnit 5 jars
-* Migrated off JUnit 3.8. This program assumes you are on JUnit 4. JUnit 3.8 code will continue to work in JUnit 5. (as will JUnit 4 code.) But the goal of this program is to migrate JUnit 4 syntax to JUnit 5 syntax.
+* Conversion off JUnit 3.8. This program assumes you are on JUnit 4. JUnit 3.8 code will continue to work in JUnit 5. (as will JUnit 4 code.) But the goal of this program is to convert JUnit 4 syntax to JUnit 5 syntax.
 
 # Running the program
 Pass the absolute path of the directory containing your tests (ex /my/dir/src/test/java) when running class jb.Updater. The program outputs the name of each file as it goes through them. It runs quickly. Under a few seconds for two thousand files.
-Then make manual edits at end (or rollback those test classes and deal with them later). In particular the following will not compile after running the program and require manually migration.
+Then make manual edits at end (or rollback those test classes and deal with them later). In particular the following will not compile after running the program and require manual conversion.
 1. Parameterized tests
-1. Tests that use expected parameter
-1. Tests that use timeout parameter
+1. Runners
+1. Rules
+1. Categories (there is only basic support for categories)
 
 This program is also available as a shaded jar. To run:
 java -jar convert-junit4-to-to-junit5-0.0.1.jar /my/dir/src/test/java
 
 # Side effects
-This program updates every class and reformats your code. This is a side effect of using [Java Parser](https://github.com/javaparser/javaparser). If you have a standard coding format, you can always reapply it after running. Regardless, make sure to commit your code before running the program so this commit only has the refactoring to JUnit 5.
+This program updates every class and reformats your code.
+This is a side effect of using [Java Parser](https://github.com/javaparser/javaparser).
+If you have a standard coding format, you can always reapply it after running. Regardless, make sure to commit your code before running the program so this commit only has the refactoring to JUnit 5.
+
+# Running the update from code
+There are [configuration options](convert-junit4-to-junit5/src/main/java/jb/configuration/Configuration.java) like `preserve formatting` and `dry run` that are not exposed as command line flags.
+Have a look at [ConvertProgrammatically](convert-junit4-to-junit5/src/main/java/jb/UpdateWithAdditionalOptions.java) to see how to enable those options from the code.
+This tool Is build around the idea of list of conversions that are executed in sequence one after the other.
```

Comment:
I re-phrased it. See if you like it.

Reply:
>>  think Java 8 is needed to run this tool, isn't it? I think this is worth mention it, because 
>> further versions of JUnit don't need Java 8
Java 8 is needed to run JUnit 5 so these seems reasonable whether it gets mentioned or not

Reply:
Added a bullet point to the Pre-reqs